pallets/ddc-validator/src/lib.rs

```diff
@@ -0,0 +1,949 @@
+	#[pallet::storage]
+	#[pallet::getter(fn signal)]
+	pub(super) type Signal<T: Config> = StorageValue<_, bool>;
+
+	/// The map from the era and CDN participant stash key to the validation decision related.
+	#[pallet::storage]
+	#[pallet::getter(fn validation_decisions)]
+	pub type ValidationDecisions<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, EraIndex, Twox64Concat, T::AccountId, ValidationDecision>;
+
+	// Map to check if validation decision was performed for the era
+	#[pallet::storage]
+	#[pallet::getter(fn validation_decision_set_for_node)]
+	pub(super) type ValidationDecisionSetForNode<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, EraIndex, Twox64Concat, T::AccountId, bool, ValueQuery>;
+
+	// Map to check if reward points were set for the era
+	#[pallet::storage]
+	#[pallet::getter(fn reward_points_set_for_node)]
+	pub(super) type RewardPointsSetForNode<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, EraIndex, Twox64Concat, T::AccountId, bool, ValueQuery>;
+
+	/// The last era for which the tasks assignment produced.
+	#[pallet::storage]
+	#[pallet::getter(fn last_managed_era)]
+	pub type LastManagedEra<T: Config> = StorageValue<_, EraIndex>;
+
+	/// The mapping of controller accounts to OCW public keys
+	#[pallet::storage]
+	#[pallet::getter(fn ocw_keys)]
+	pub type OffchainWorkerKeys<T: Config> =
+		StorageMap<_, Twox64Concat, T::AccountId, T::AccountId>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		NotController,
+		OCWKeyNotRegistered,
+		ContentOwnersDoubleSpend,
+		ValidationDecisionAlreadySet,
+		NodeNotActive,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config>
+	where
+		<T as frame_system::Config>::AccountId: AsRef<[u8]> + UncheckedFrom<T::Hash>,
+		<BalanceOf<T> as HasCompact>::Type: Clone + Eq + PartialEq + Debug + TypeInfo + Encode,
+	{
+		// Validator submits decision for an era
+		ValidationDecision(EraIndex, T::AccountId, ValidationDecision),
+		// Set era reward points
+		EraRewardPoints(EraIndex, Vec<(T::AccountId, u64)>),
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T>
+	where
+		<T as frame_system::Config>::AccountId: AsRef<[u8]> + UncheckedFrom<T::Hash>,
+		<BalanceOf<T> as HasCompact>::Type: Clone + Eq + PartialEq + Debug + TypeInfo + Encode,
+	{
+		fn on_initialize(block_number: T::BlockNumber) -> Weight {
+			if block_number <= 1u32.into() {
+				return Weight::from_ref_time(0)
+			}
+
+			Signal::<T>::set(Some(false));
+
+			let current_ddc_era = match ddc_staking::pallet::Pallet::<T>::current_era() {
+				Some(era) => era,
+				None => {
+					log::debug!("DDC era not set.");
+					return Weight::from_ref_time(0)
+				},
+			};
+			log::debug!("Current DDC era: {:?}.", current_ddc_era);
+
+			// Produce an assignment for the next era if it's not produced yet.
+			match Self::last_managed_era() {
+				Some(last_managed_era) if current_ddc_era < last_managed_era =>
+					return Weight::from_ref_time(0),
```

Comment:
why is weight returned?

Reply:
Check https://paritytech.github.io/substrate/master/frame_support/traits/trait.Hooks.html#method.on_initialize